astropy/io/votable/tree.py

```diff
@@ -1675,7 +1677,14 @@ def parse(self, iterator, config):
     def to_xml(self, w, **kwargs):
         attrib = w.object_attrs(self, self._attr_list)
         if "unit" in attrib:
-            attrib["unit"] = self.unit.to_string("cds")
+            format = _get_unit_format(self._config)
+            try:
+                attrib["unit"] = self.unit.to_string(format)
+            except ValueError as e:
+                # Allow non-standard units with a warning, see
+                # https://github.com/astropy/astropy/issues/17497#issuecomment-2520472495
+                attrib["unit"] = self.unit.to_string()
+                warnings.warn(str(e), u.UnitsWarning)
```

Comment:
@ManonMarchand @msdemlei -- Do you have opinion whether this should be `UnitsWarning` or [W50](https://docs.astropy.org/en/latest/io/votable/api_exceptions.html#w50-invalid-unit-string-x)?

Reply:
Update: I changed it to W50, as requested.